### ISSUE TITLE
Fail2ban doesn't work for log entries originating from localhost 127.0.0.1

### ISSUE THREAD

**Issue by eayin2** [fail2ban/fail2ban#2322]
### Environment:

- Fail2Ban version (including any possible distribution suffixes): v0.10.2
- OS, including release name/version: Ubuntu 18.04
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:

Fail2ban doesn't notice alerts originating from 127.0.0.1 (localhost). I use a custom filter and action. My Filter gets triggered from LAN successfully, just not from localhost.
`ssh root@localhost -i my_key -p 12944` <-- does not trigger alert
`ssh root@192.168.10.10 -i my_key -p 12944` <--- triggers alert

`fail2ban-regex` verified that my filter matches `Jan 14 07:39:41 myhost01 sshd[26433]: Connection closed by authenticating user root 127.0.0.1 port 56216 [preauth]` 68 times inside `auth.log`, but `fail2ban-client status sshd-preauth` shows 0 matched filters and actions. What might I be missing?

I set `ignoreself = false` and `ignoreip =` to nothing, but it's still ignoring 127.0.0.1 log entries.

#### jail.d/custom.local
```
[sshd-preauth]

enabled = true
logpath = /var/log/auth.log
backend = auto
banaction = disable-user
maxretry = 1
findtime = 600
filter  = sshd-preauth
```

#### filter.d/sshd-preauth.conf
```
[INCLUDES]
before = common.conf

[DEFAULT]

_daemon = sshd
__pref = (?:(?:error|fatal): (?:PAM: )?)?
__suff = (?: \[preauth\])?\s*
__on_port_opt = (?: port \d+)?(?: on \S+(?: port \d+)?)?
__alg_match = (?:(?:\w+ (?!found\b)){0,2}\w+)

[Definition]
prefregex = ^<F-MLFID>%(__prefix_line)s</F-MLFID>%(__pref)s<F-CONTENT>.+</F-CONTENT>$
failregex = ^Connection closed by authenticating user root (?P<ip4>(?:\d{1,3}\.){3}\d{1,3}) port \d+ \[preauth\]$
datepattern = {^LN-BEG}
```

#### action.d/disable-user.conf
```
[Definition]
actionstart = touch /tmp/01234
actionstop =
actioncheck = 
actionban = touch /tmp/OSSEC
actionunban = touch /tmp/UNBAN
```

#### fail2ban-regex
```
fail2ban-regex /var/log/auth.log /etc/fail2ban/filter.d/sshd-preauth.conf  

Running tests

Use   failregex filter file : sshd-preauth, basedir: /etc/fail2ban
Use      datepattern : Default Detectors
Use         log file : /var/log/auth.log
Use         encoding : UTF-8


Results

Failregex: 68 total
|-  #) [# of hits] regular expression
|   1) [68] ^Connection closed by authenticating user root (?P<ip4>(?:\d{1,3}\.){3}\d{1,3}) port \d+ \[preauth\]$
`-

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [865] {^LN-BEG}(?:DAY )?MON Day %k:Minute:Second(?:\.Microseconds)?(?: ExYear)?
`-

Lines: 865 lines, 0 ignored, 68 matched, 797 missed
[processed in 0.06 sec]

Missed line(s): too many to print.  Use --print-all-missed to print all 797 lines
```

`jail.conf`
```
[DEFAULT] 
ignorself = false
ignoreip =
```

Is this a known limitation for fail2ban or is there a workaround?

**Comment by sebres** [comment]
This was a bug (typo) in the jail.conf template definition, see #2198.
Just rename `ignorself` into `ignoreself` in your jail.local.

BTW, specifying this in `[DEFAULT]` section may cause unexpected ban... Possibly the jail (you want it explicitly) instead of default would be better here.

**Comment by eayin2** [comment]
I renamed it to `ignoreself`, yet I have the same error pattern. I suspect that ignoring the localhost is hardcoded to fail2ban. Correct me if I'm wrong.

**Comment by sebres** [comment]
> Correct me if I'm wrong.

You are.

The `jail.local` should contain:
```ini
[DEFAULT]
ignoreself = false
# or per jail (if in default it is true/not specified):
[jail-you-want-test]
ignoreself = false
```
I'm pretty sure `ignoreip` parameter is empty per default (was replaced with newest `ignoreself`).

You should restart/reload fail2ban in order to update the runtime parameters.
Otherwise you can do it per jail (without restart)... `fail2ban-client set "<jail>" ignoreself false`.

Additionally there were some 0.10th versions with another bug as regards the `ignoreself` (see https://github.com/fail2ban/fail2ban/issues/2078#issuecomment-384670619), which was fixed in 60b36c421380b1ded02c5c6118f9e6dda070249d.

Note also, `fail2ban-regex` does not consider this parameter at all (it shows matches/misses only).

**Comment by eayin2** [comment]
Indeed, with the newest release 0.10.4 (October 2018) it works. It didn't work with v0.10.2. Thanks for the help!
